activejob/lib/active_job/queue_adapters/resque_adapter.rb

```diff
@@ -27,6 +27,7 @@ module QueueAdapters
     #   Rails.application.config.active_job.queue_adapter = :resque
     class ResqueAdapter
       def enqueue(job) #:nodoc:
+        JobWrapper.instance_variable_set(:@queue, job.queue_name)
```

Comment:
We should change `JobWrapper` to be an instance and set this at initialization time. Also could you add a test to this behavior?